### PR TITLE
Add encryption option and components for PostgreSQL, MySQL, MongoDB

### LINKED ISSUE #145: Encryption at-rest add-on option
Some add-ons will need an encryption at-rest option. Those add-ons are, for now:
- PostgreSQL
- MySQL
- MongoDB

## Encryption at-rest option

The encryption at-rest option is an option to encrypt at-rest the disk of the add-on.

### Description

- What it does
- The price (Free for now. 10% of the plan price when we will start to bill it)
- A link to our documentation, for example https://www.clever-cloud.com/doc/addons/postgresql/#encryption-at-rest

## Create the corresponding `<cc-addon-postgresql-options>`, `<cc-addon-mysql-options>`, `<cc-addon-mongodb-options>`

Each add-on will have the same option, only the documentation link will change.

The option will be disabled by default.

### PR DESCRIPTION
**Pull request by BlackYoup**
Fixes #145

### CODE REVIEW

File: src/templates/cc-addon-encryption-at-rest-option.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import '../molecules/cc-error.js';
+import { html } from 'lit-element';
+import { assetUrl } from '../lib/asset-url.js';
+import { i18n } from '../lib/i18n.js';
+
+const LOGO = assetUrl(import.meta, '../assets/encryption-at-rest.svg');
```
Comment (hsablonniere) [inline src/templates/cc-addon-encryption-at-rest-option.js:6]: The current convention is `camelCaseNameOfTheImageSvg` with a `Svg` suffix.

File: src/addon/cc-addon-postgresql-options.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  _onFormOptionsSubmit ({ detail }) {
+    dispatchCustomEvent(this, 'submit', detail);
+  }
+
+  _getFormOptions () {
+    return this.options.map((option) => {
+      switch (option) {
+        case 'encryption': return ccAddonEncryptionAtRestOption(ENCRYPTION_DOCUMENTATION_URL, this.encryptionPrice);
+        default: return null;
```
Comment (hsablonniere) [inline src/addon/cc-addon-postgresql-options.js:55]: Not a great fan of one liners like this. Is ESLint OK with that? If so, let's keep it and I'll update the ESLint config :p

File: src/addon/cc-addon-postgresql-options.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      };
+    }).filter((option) => option !== null);
```
Comment (hsablonniere) [inline src/addon/cc-addon-postgresql-options.js:57]: In those situations, you can align the `.map` and `.filter`

File: src/addon/cc-addon-mysql-options.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+ * @prop {Number} encryptionPrice - Price of the encryption option.
+ * @prop {String[]} options - List of options to enable for this selection.
+ *
+ * @event {CustomEvent<Options>} cc-addon-mysql-options:submit - Fires when the form is submitted.
+ */
+export class CcAddonMysqlOptions extends LitElement {
+
+  static get properties () {
+    return {
+      options: { type: Array, attribute: 'options' },
```
Comment (hsablonniere) [inline src/addon/cc-addon-mysql-options.js:35]: No need for `attribute` there.

File: src/addon/cc-addon-mongodb-options.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+ * @prop {Number} encryptionPrice - Price of the encryption option.
+ * @prop {String[]} options - List of options to enable for this selection.
+ *
+ * @event {CustomEvent<Options>} cc-addon-mongodb-options:submit - Fires when the form is submitted.
+ */
+export class CcAddonMongodbOptions extends LitElement {
+
+  static get properties () {
+    return {
+      options: { type: Array, attribute: 'options' },
```
Comment (hsablonniere) [inline src/addon/cc-addon-mongodb-options.js:35]: No need for `attribute` here.

File: src/addon/cc-addon-postgresql-options.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+ * @prop {Number} encryptionPrice - Price of the encryption option.
+ * @prop {String[]} options - List of options to enable for this selection.
+ *
+ * @event {CustomEvent<Options>} cc-addon-postgresql-options:submit - Fires when the form is submitted.
+ */
+export class CcAddonPostgresqlOptions extends LitElement {
+
+  static get properties () {
+    return {
+      options: { type: Array, attribute: 'options' },
```
Comment (hsablonniere) [inline src/addon/cc-addon-postgresql-options.js:35]: No need for `attribute` here.

**Review by hsablonniere — CHANGES_REQUESTED** [review]
I haven't looked at the result in a browser yet.

File: src/addon/cc-addon-postgresql-options.js (hunk 2 above)
Comment (BlackYoup) [reply src/addon/cc-addon-postgresql-options.js:55]: Yes, eslint is ok :p

File: src/templates/cc-addon-encryption-at-rest-option.js (hunk 1 above)
Comment (BlackYoup) [reply src/templates/cc-addon-encryption-at-rest-option.js]: Should be fixed now

File: src/addon/cc-addon-postgresql-options.js (hunk 3 above)
Comment (BlackYoup) [reply src/addon/cc-addon-postgresql-options.js]: Updated, please double check to see if I correctly understood you

File: src/addon/cc-addon-postgresql-options.js (hunk 2 above)
Comment (hsablonniere) [reply src/addon/cc-addon-postgresql-options.js]: OK so let's keep it like that, we'll update ESLint later.

File: src/addon/cc-addon-postgresql-options.js (hunk 3 above)
Comment (hsablonniere) [reply src/addon/cc-addon-postgresql-options.js]: This is what I expected.

**Comment by hsablonniere** [comment]
I didn't understood at first that options were opt-in with `options`.
We need a story to see what the component looks like when `options` is an empty array.
Will we need an empty message?

I see `encryptionEnabled` in:

* the JSDoc header
* the `static get properties () {}`
* the `constructor`

but it's never used in the component template and never used by the story.

**Comment by BlackYoup** [comment]
> I didn't understood at first that options were opt-in with options.

Yes, options might be different across add-ons plans

> We need a story to see what the component looks like when options is an empty array.

Yeah, I didn't make one because I assumed that you need to provide at least one option, otherwise this option selection shouldn't even be displayed. I can indeed add a `there are no options` text if you think it's worth adding such a case.

I fixed the `encryptionEnabled` boolean not correctly displayed and added a story with the , sorry for the silly mistake.

**Comment by hsablonniere** [comment]
> This option is currently free but will then be billed 10.0% of the plan's price. If this option is enabled, it will be billed €10.00 per month.

I still think it's a bit confusing.

**Comment by hsablonniere** [comment]
About the translation, @urcadox suggests:

> This option is currently free. In the future, it will be billed 10% of the plan price, which amounts to €10 per month here.

**Comment by hsablonniere** [comment]
I think we should move `options` from `String[]` to `Option[]` where `Option` has a `name`, a `enabled` boolean and some custom options like `price` for encryption or `flavor` for apm and kibana.

**Pull request merged**